Source/DafnyCore/DafnyMain.cs

```diff
@@ -319,6 +319,12 @@ private class IncludeComparer : IComparer<Include> {
       Microsoft.Boogie.Program boogieProgram, string programId) {
       var moduleId = (programId ?? "main_program_id") + "_" + moduleName;
 
+      var proverPath = DafnyOptions.O.ProverOptions.Find(o => o.StartsWith("PROVER_PATH="));
+      if (proverPath is null && DafnyOptions.O.Verify) {
+        Console.WriteLine("Z3 not found in explicit --solver-path option, Dafny distribution, or PATH environment variable.");
```

Comment:
```suggestion
        Console.WriteLine("Z3 not found. Please either provide a path to the z3 executable using the `--solver-path <path>` option, or manually place the `z3` folder next to the Dafny executable you are using (this folder should contain `bin/z3` or `bin/z3.exe`), or set the PATH environment variable to also target the folder containing the z3 executable.");
```
Errors with explicit action items are always appreciated.

Reply:
I updated the text to a slightly reworded version of your suggestion.